lib/rspec/matchers/dsl.rb

```diff
@@ -80,11 +77,20 @@ def match(&match_block)
         # @yield [Object] actual the actual value (i.e. the value wrapped by `expect`)
         def match_when_negated(&match_block)
           define_user_override(:does_not_match?, match_block) do |actual|
-            @actual = actual
-            super(*actual_arg_for(match_block))
+            begin
+              @actual = actual
+              RSpec::Support.with_failure_notifier(RAISE_METHOD) do
+                super(*actual_arg_for(match_block))
+              end
+            rescue RSpec::Expectations::ExpectationNotMetError
+              false
+            end
           end
         end
 
+        # @private
+        RAISE_METHOD = method(:raise)
+
```

Comment:
I moved this because it seems more appropriate to place after both `#match` and `#match_when_negated` (it was put between them originally because it was only used in `#match`). Although I think it makes more sense here now, if your intention is just to keep the lines in git history belonging to the original commit where functionality was changed, I understand. I'll move that back.